### ISSUE TITLE
Support optional parameters in blazor routing

### ISSUE THREAD

**Issue by ichensky** [dotnet/aspnetcore#10759]
Routing with nulable types not working. 
1. Create blazor page.
2. Add route  'fetchdata/{page:int?}'
There will be an exception: 
```
Invalid template 'fetchdata/{page:int?}'. 
The character '?' in parameter segment '{page:int?}' is not allowed.
```
Also it is not possible create "double routing" like in the issue #10686

**Comment by SteveSandersonMS** [comment]
Note: when we do this, we need to make sure it supplies `null` for any omitted optional parameters for consistency with https://github.com/aspnet/AspNetCore/pull/12407.

**Comment by captainsafia** [comment]
Closing this as it has been implemented in #19733.
